gnmi_server/gnoi.go

```diff
@@ -20,6 +20,46 @@ import (
 	jwt "github.com/dgrijalva/jwt-go"
 )
 
+func KillOrRestartProcess(restart bool, serviceName string) error {
+	sc, err := ssc.NewDbusClient()
+	if err != nil {
+		return err
+	}
+	if restart {
+		log.V(2).Infof("Restarting service %s...", serviceName)
+		err = sc.RestartService(serviceName)
+		if err != nil {
+			log.V(2).Infof("Failed to restart service %s: %v", serviceName, err)
+		}
+	} else {
+		log.V(2).Infof("Stopping service %s...", serviceName)
+		err = sc.StopService(serviceName)
+		if err != nil {
+			log.V(2).Infof("Failed to stop service %s: %v", serviceName, err)
+		}
+	}
+	return err
+}
+
+func (srv *Server) KillProcess(ctx context.Context, req *gnoi_system_pb.KillProcessRequest) (*gnoi_system_pb.KillProcessResponse, error) {
+	serviceName := req.GetName()
+	restart := req.GetRestart()
+        if req.GetPid() != 0 {
+            return nil, status.Errorf(codes.Unimplemented, "Pid option is not implemented")
+        }
+        if req.GetSignal() != gnoi_system_pb.KillProcessRequest_SIGNAL_UNSPECIFIED {
+            return nil, status.Errorf(codes.Unimplemented, "Signal option is not implemented")
+        }
+	log.V(1).Info("gNOI: KillProcess with optional restart")
+	log.V(1).Info("Request:", req)
```

Comment:
Does this log work? Do you have sample output?

Reply:
Yes.

Sample log:
```
2024 May 22 01:31:51.650270 str-s6100-acs-1 INFO gnmi#supervisord: gnmi-native I0522 01:31:51.649487      27 gnoi.go:151] gNOI: Time
2024 May 22 01:32:17.224515 str-s6100-acs-1 INFO gnmi#supervisord: gnmi-native I0522 01:32:17.223930      27 gnoi.go:47] gNOI: KillProcess with optional restart
2024 May 22 01:32:17.224748 str-s6100-acs-1 INFO gnmi#supervisord: gnmi-native I0522 01:32:17.223964      27 gnoi.go:48] Request:name:"swss"  restart:true
2024 May 22 01:32:17.224920 str-s6100-acs-1 INFO gnmi#supervisord: gnmi-native I0522 01:32:17.224042      27 gnoi.go:29] Restarting service swss...
```